### PR TITLE
Rule-modal functionality

### PR DESCRIPTION
**Pull request by mikuweb**
- Create Rule component
- Open modal when the user's first visit the page
- Add animation to the modal

https://github.com/mikuweb/Pokemon-App/assets/100392926/d5073232-747a-428e-af4f-9a7224eb28e8

### CODE REVIEW

**Comment by Escapado** [comment]
The text in the modal for each rule is centered but the bullet-points are left-aligned. I would suggest you either
a) Remove the bullet points with the tailwind class: "list-none"
b) Make the text left-aligned

File: src/features/Rule/index.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import { Dialog, Transition } from "@headlessui/react";
+import React, { Fragment, useEffect, useState } from "react";
+import { AiOutlineClose } from "react-icons/ai";
+
+const Rule = () => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  // OPEN RULE-MODAL USER'S FIRST VISIT
```
Comment (Escapado) [inline src/features/Rule/index.tsx]: I would say you should show the modal the first time the user plays. If the user played this before I don't think it's necessary to show the modal each time they play again. I would suggest to save in local storage whether the user has already played the game and as that as a check before showing the modal.

**Review by Escapado — APPROVED** [review]

**Comment by mikuweb** [comment]
>The text in the modal for each rule is centered but the bullet-points are left-aligned. I would suggest you either
a) Remove the bullet points with the tailwind class: "list-none"
b) Make the text left-aligned

Thank you! I made the text left-aligned.
Also I fixed some styles in the same commit.

Commit: [fix(style): make the text left-aligned](https://github.com/mikuweb/Pokemon-App/pull/3/commits/1449b7d92b669a6b22dfd521314371aebe4f4ea0)

File: src/features/Rule/index.tsx (hunk 1 above)
Comment (mikuweb) [reply src/features/Rule/index.tsx]: I added localStrage to save and check the user's play history.
Do you think there are some nice way to refactor the code as I use `const hasPlayed = localStorage.getItem("hasPlayed");` twice in multiple files?
I was going to export this but I couldn't..

Commit: [feat: check if the user has played before](https://github.com/mikuweb/Pokemon-App/pull/3/commits/79e6243950fb6f08ee9aabacb718813d832a8282)

**Comment by mikuweb** [comment]
@Escapado 
The error occurred when it's Building but do you know why?

File: src/features/Rule/index.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  const [isOpen, setIsOpen] = useState(false);
+
+  // OPEN RULE-MODAL: USER'S FIRST VISIT
+  useEffect(() => {
+    const hasPlayed = localStorage.getItem("hasPlayed");
+
+    if (!hasPlayed) {
+      setTimeout(() => {
+        setIsOpen(true);
+        localStorage.setItem("hasPlayed", "true");
```
Comment (Escapado) [inline src/features/Rule/index.tsx]: This is redundant.

**Review by Escapado — APPROVED** [review]

File: src/features/Rule/index.tsx (hunk 2 above)
Comment (mikuweb) [reply src/features/Rule/index.tsx]: Thanks, I removed it.
https://github.com/mikuweb/Pokemon-App/pull/3/commits/3df8c5334b5ebb4dd80ecc1de26bc271b6ef8efc

**Comment by mikuweb** [comment]
Memo:
>The error occurred when it's Building but do you know why?

It's fixed bu updating libraries. https://github.com/mikuweb/Pokemon-App/pull/3/commits/f313d03cfa528ac98522ec96e32f8bcb63432677

https://github.com/tailwindlabs/headlessui/issues/2479

**Pull request merged**